File: app/src/main/java/com/hfad/myferma/incubator/MenuAdapter.java
package com.hfad.myferma.incubator;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

public class MenuAdapter  extends FragmentStateAdapter {
    public MenuAdapter(@NonNull FragmentActivity fragmentActivity) {
        super(fragmentActivity);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        switch (position){
            case 0:
                return new HomeIncubatorFragment();
            case 1:
                return new ArchiveIncubatorFragment();
            default:
                return new HomeIncubatorFragment();
        }

    }

    @Override
    public int getItemCount() {
        return 2;
    }
}

File: app/src/main/java/com/hfad/myferma/incubator/MenuIncubator/MenuAdapter.kt
package com.hfad.myferma.incubator.MenuIncubators

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter

class MenuAdapter constructor(fragmentActivity: FragmentActivity) :
    FragmentStateAdapter(fragmentActivity) {
    override fun createFragment(position: Int): Fragment {
        return when (position) {
            0 -> HomeIncubatorFragment()
            1 -> ArchiveIncubatorFragment()
            else -> HomeIncubatorFragment()
        }
    }

    override fun getItemCount(): Int {
        return 2
    }
}